### PR TITLE
Rename sanitycheck -> twister

### PR DESCRIPTION
**Pull request by nashif**
This collection of commits renames sanitycheck script and anything related to that to the new name:

# twister #

Reason for this rename is to comply with inclusive language guidelines.

For reference: https://en.wikipedia.org/wiki/Tornado

### CODE REVIEW

**Review by galak — APPROVED** [review]

**Review by ioannisg — APPROVED** [review]

**Review by tyrantlizard77 — COMMENTED** [review]
consider adding a deprecated "sanitycheck" wrapper script which just prints a warning "This was renamed to twister" and invokes twister with the provided args, so that users aren't taken completely by surprise when the script seems to no longer exist. and then remove the placeholder script one release later (2.6 LTS). just a thought.

File: .buildkite/run.sh [hunk 1]
```diff
@@ -6,13 +6,13 @@ set -eE
...
-	   touch sanity-out/sanitycheck.xml
+	# Rename twister junit xml for use with junit-annotate-buildkite-plugin
+	# create dummy file if twister did nothing
+	if [ ! -f sanity-out/twister.xml ]; then
+	   touch sanity-out/twister.xml
 	fi
-	mv sanity-out/sanitycheck.xml sanitycheck-${BUILDKITE_JOB_ID}.xml
-	buildkite-agent artifact upload sanitycheck-${BUILDKITE_JOB_ID}.xml
+	mv sanity-out/twister.xml twister-${BUILDKITE_JOB_ID}.xml
+	buildkite-agent artifact upload twister-${BUILDKITE_JOB_ID}.xml
```
Comment (jenmwms) [inline .buildkite/run.sh:16]: Are these remaining `sanity-out` intentional, or does it need to be renamed too?

File: .gitignore [hunk 2]
```diff
@@ -32,6 +32,7 @@ doc/samples
 doc/latex
 doc/themes/zephyr-docs-theme
 sanity-out*
```
Comment (jenmwms) [inline .gitignore:34]: Is this remaining `sanity-out` intentional, or does it need to be renamed too?

File: doc/guides/coverage.rst [hunk 3]
```diff
@@ -132,24 +132,24 @@ You may postprocess these with your preferred tools. For example:
...
 
-    $ sanitycheck --coverage -p qemu_x86 -T tests/kernel
+    $ twister --coverage -p qemu_x86 -T tests/kernel
 
 or::
 
-    $ sanitycheck --coverage -p native_posix -T tests/bluetooth
+    $ twister --coverage -p native_posix -T tests/bluetooth
 
 which will produce ``sanity-out/coverage/index.html`` with the report.
```
Comment (jenmwms) [inline doc/guides/coverage.rst:147]: Does this need to be renamed too?

File: scripts/ci/run_ci.sh [hunk 4]
```diff
@@ -94,14 +94,14 @@ function on_complete() {
 	mkdir -p shippable/testresults
 	mkdir -p shippable/codecoverage
 
-	if [ -e ./sanity-out/sanitycheck.xml ]; then
-		echo "Copy ./sanity-out/sanitycheck.xml"
-		cp ./sanity-out/sanitycheck.xml shippable/testresults/
+	if [ -e ./sanity-out/twister.xml ]; then
+		echo "Copy ./sanity-out/twister.xml"
+		cp ./sanity-out/twister.xml shippable/testresults/
 	fi
```
Comment (jenmwms) [inline scripts/ci/run_ci.sh:100]: Similar to .buildkite/run.sh - does it need to be renamed too?

**Review by yerabolu — COMMENTED** [review]
report_name in needs to be changed as well to address all the report names.

**Comment by yerabolu** [comment]
> report_name in needs to be changed as well to address all the report names.

Sorry, ignore this comment. changes to report names are already in.

**Review by jenmwms — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

File: doc/contribute/index.rst [hunk 5]
```diff
@@ -293,23 +293,23 @@ Note, gitlint only checks HEAD (the most recent commit), so you should run it
 after each commit, or use the ``--commits`` option to specify a commit range
 covering all the development patches to be submitted.
 
-sanitycheck
+twister
 ===========
```
Comment (tejlmand) [inline doc/contribute/index.rst:297]: Need to shorten this:
```suggestion
=======
```

**Comment by nashif** [comment]
> consider adding a deprecated "sanitycheck" wrapper script which just prints a warning "This was renamed to twister" and invokes twister with the provided args, so that users aren't taken completely by surprise when the script seems to no longer exist. and then remove the placeholder script one release later (2.6 LTS). just a thought.

done

File: doc/contribute/index.rst [hunk 6]
```diff
@@ -293,23 +293,23 @@ Note, gitlint only checks HEAD (the most recent commit), so you should run it
...
 covering all the development patches to be submitted.
 
-sanitycheck
-===========
+twister
+=======
 
 .. note::
-   sanitycheck does not currently run on Windows.
+   twiser does not currently run on Windows.
```
Comment (MaureenHelm) [inline doc/contribute/index.rst:300]: s/twiser/twister/

**Review by MaureenHelm — COMMENTED** [review]
Looks good, just one nit

**Review by tejlmand — APPROVED** [review]

**Pull request merged**